cachecontrol/cache.py

```diff
@@ -7,37 +7,39 @@
 safe in-memory dictionary.
 """
 from threading import Lock
+from typing import TYPE_CHECKING, MutableMapping, Optional
 
+if TYPE_CHECKING:
+    from io import BufferedIOBase
 
-class BaseCache(object):
 
-    def get(self, key):
+class BaseCache(object):
+    def get(self, key: str) -> Optional[bytes]:
         raise NotImplementedError()
 
-    def set(self, key, value, expires=None):
+    def set(self, key: str, value: bytes, expires: Optional[int] = None) -> None:
         raise NotImplementedError()
 
-    def delete(self, key):
+    def delete(self, key: str) -> None:
         raise NotImplementedError()
 
-    def close(self):
+    def close(self) -> None:
         pass
 
 
 class DictCache(BaseCache):
-
-    def __init__(self, init_dict=None):
+    def __init__(self, init_dict: Optional[MutableMapping[str, bytes]] = None) -> None:
         self.lock = Lock()
         self.data = init_dict or {}
 
-    def get(self, key):
+    def get(self, key: str) -> Optional[bytes]:
         return self.data.get(key, None)
 
-    def set(self, key, value, expires=None):
+    def set(self, key: str, value: bytes, expires: Optional[int] = None) -> None:
```

Comment:
`expires` should also accept a datetime object, see the RedisCache implementation